src/components/AuthenticationModal.jsx

```diff
@@ -8,23 +8,41 @@ import Login from './Login';
 import ShowModal from '../store/actions/changeFormAction';
 import { showErrors } from '../store/actions/registerActions';
 import { IS_LOADING } from '../store/actions/actionTypes';
+import InitiateResetForm from './InitateResetForm';
+import PasswordResetForm from './PasswordResetForm';
 
 export class ConnectedAuthenticationModal extends React.Component {
   render() {
-    const { modalShow, dispatch, isRegister } = this.props;
+    const { modalShow, dispatch, component } = this.props;
     const onHide = () => {
       dispatch(ShowModal({ modalShow: false }));
       dispatch({ type: IS_LOADING, payload: { isLoading: false } });
       dispatch(showErrors({}));
     };
-    let Form;
+    let CurrentComponent;
     let headerMessage;
-    if (isRegister) {
-      Form = RegisterForm;
-      headerMessage = <h3 className="col-md-10 col-md-offset-5">Join Author&apos;s Haven</h3>;
-    } else {
-      Form = Login;
-      headerMessage = <h3 className="col-md-10 col-md-offset-5">Welcome back</h3>;
+
+    switch (component) {
+      case ('register'):
+        CurrentComponent = RegisterForm;
+        headerMessage = <h3 className="col-md-10 col-md-offset-5">Join Author&apos;s Haven</h3>;
+        break;
+      case ('login'):
+        CurrentComponent = Login;
+        headerMessage = <h3 className="col-md-10 col-md-offset-5">Welcome back</h3>;
+        break;
+      case ('initiate-reset'):
+        CurrentComponent = InitiateResetForm;
+        headerMessage = <h3 className="col-md-10 col-md-offset-5">Forgot your password?</h3>;
+        break;
+      case ('password-reset'):
+        CurrentComponent = PasswordResetForm;
+        headerMessage = <h3 className="col-md-10 col-md-offset-5">Create your new Password</h3>;
+        break;
+      default:
+        CurrentComponent = Login;
+        headerMessage = <h3 className="col-md-10 col-md-offset-5">Welcome back</h3>;
+        break;
```

Comment:
@ElMonstro, it would be best to have messages in strings for this switch block instead of `<h3 className="col-md-10 col-md-offset-5">` which is repeated all over, you can parse the string then in `<h3 className="col-md-10 col-md-offset-5">` later.